src/Elliptic/FirstOrderOperator.hpp

```diff
@@ -60,27 +60,29 @@ struct FirstOrderSourcesImpl<Dim, tmpl::list<PrimalFields...>,
       const Variables<db::wrap_tags_in<
           ::Tags::Flux, VarsTags, tmpl::size_t<Dim>, Frame::Inertial>>& fluxes,
       const SourcesArgs&... sources_args) noexcept {
+    // Compute primal sources. They may depend on the auxiliary variables in
+    // addition to the primal variables. However, we pass the fluxes instead of
+    // the auxiliary variables to the source computer as an optimization so they
+    // don't have to be re-computed.
+    sources->initialize(vars.number_of_grid_points(), 0.);
```

Comment:
[optional and depends on a lot of other generic code] you may want to document whether systems can assume this to be true. In the evolution code we do not guarantee inputs to be zeroed. I think either design is reasonable, but it might be good to document this guarantee for users. Of course, that documentation might be better placed in a tutorial later on :)

Reply:
Yea I'll definitely need to document this sort of thing in a tutorial on writing elliptic systems. I'll also add it to the docs for the DG operator that will replace this function.